test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,459 @@
+7) create and configure the custom CoreDNS configMap with the IP received above.
+8) Create custom CoreDNS deployment.
+9) Create and apply antrea FQDN policy.
+10) Deploy antrea-toolbox.
+
+
+---------- tic
+11) curl the FQDN from within toolbox.
+12) imitate caching the IP belonging to above FQDN resolution by keeping it in a variable.
+13) edit configmap with the other IP.
+14) wait for new IP to get updated in configMap and let the changes be reflected in dns pod.
+15) curl the FQDN again with IP , simulating usage of cache -- and it must fail with no connectivity.
+*/
+
+func TestFQDNPolicyWithCachedDNS(t *testing.T) {
+	skipIfAntreaPolicyDisabled(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+
+	testFqdn := "nginx-test-pod.lfx.test"
+
+	//TODO: Check for IPv6 ?
+	ipFamily := v1.IPv4Protocol
+
+	// Create the service .
+	//TODO: Should the names be put up as constants instead of direct strings here?
+	customDnsService, err := testData.CreateUDPService("custom-dns-service", data.testNamespace, 53, 53, map[string]string{"app": "custom-dns"}, false, false, v1.ServiceTypeClusterIP, &ipFamily)
+	if err != nil {
+		t.Fatalf("Error when creating custom DNS service: %v", err)
+	}
+	require.NoError(t, err)
+
+	// get the IP
+	customCoreDnsServiceObject, err := data.clientset.CoreV1().Services(data.testNamespace).Get(context.Background(), customDnsService.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error when getting custom DNS service object : %v", err)
+	}
+	require.NoError(t, err)
+
+	// Print the ClusterIP
+	t.Logf("ClusterIP of the service: %s\n", customCoreDnsServiceObject.Spec.ClusterIP)
+
+	// Get Antrea ConfigMap
+	cm, err := data.GetAntreaConfigMap(antreaNamespace)
+	if err != nil {
+		t.Fatalf("Error when getting custom DNS configMap : %v", err)
+	}
+	require.NoError(t, err)
+
+	// Read current value of dnsServer
+	var agentConf agentconfig.AgentConfig
+
+	if err := yaml.Unmarshal([]byte(cm.Data["antrea-agent.conf"]), &agentConf); err != nil {
+		t.Fatalf("failed to unmarshal Agent config from ConfigMap: %v", err)
+	}
+	require.NoError(t, err)
+
+	//Set up customDNS server IP in Antrea configmap.
+	agentChanges := func(config *agentconfig.AgentConfig) {
+		config.DNSServerOverride = customCoreDnsServiceObject.Spec.ClusterIP
+	}
+	err = data.mutateAntreaConfigMap(nil, agentChanges, false, true)
+	if err != nil {
+		t.Fatalf("Error when setting up customDNS server IP in Antrea configmap : %v", err)
+	}
+
+	cm2, err := data.GetAntreaConfigMap(antreaNamespace)
+	if err != nil {
+		t.Fatalf("Error when getting custom DNS configMap : %v", err)
+	}
+	require.NoError(t, err)
+
+	// Read current value of dnsServer
+	var agentConfChanged agentconfig.AgentConfig
+	if err := yaml.Unmarshal([]byte(cm2.Data["antrea-agent.conf"]), &agentConfChanged); err != nil {
+		t.Fatalf("failed to unmarshal Agent config from ConfigMap: %v", err)
+	}
+	require.NoError(t, err)
+
```

Comment:
L73 to L108 should be a util function of its own. Also the second read as of now is completely redundant: it does not even verify that the dns server value is set to the desired one